File: 1 – FMI Data Structures & Algorithms/04–week/04–12 Reverse linked list–1 (iterative).java
// https://www.hackerrank.com/challenges/reverse-a-linked-list/problem

public static SinglyLinkedListNode reverse(SinglyLinkedListNode head) {
    SinglyLinkedListNode traverse = head;
    SinglyLinkedListNode prev = null;

    while (traverse != null) {
        SinglyLinkedListNode next = traverse.next;

        traverse.next = prev;
        prev = traverse;
        traverse = next;
    }

    return prev;
}

File: 1 – FMI Data Structures & Algorithms/04–week/04–12 Reverse linked list–1 (iterative).cpp
// https://www.hackerrank.com/challenges/reverse-a-linked-list/problem

SinglyLinkedListNode *reverse(SinglyLinkedListNode *head) {
    SinglyLinkedListNode *traverse = head;
    SinglyLinkedListNode *prev = nullptr;
    
    while (traverse != nullptr) {
        SinglyLinkedListNode *next = traverse->next;

        traverse->next = prev;
        prev = traverse;
        traverse = next;
    }
    
    return prev;
}
